SwiftDungeon/AttackButton.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  AttackButton.swift
+//  SwiftDungeon
+//
+//  Created by Puntillo Andrew J. on 3/19/18.
+//  Copyright © 2018 Toro Juan D. All rights reserved.
+//
+
+import Foundation
+import SpriteKit
+
+class AttackButton {
+    
+    //Sprite of the button
+    let buttonSprite = SKSpriteNode(imageNamed: "attack_button")
+    //Weak ref to scene
+    weak var scene: GameScene?
```

Comment:
You wont need a reference to the scene with this class. Since it will likely only be used inside something that does already though composition. ei. where you create the AttackButton instance, you can just go addChild() since its likely the game scene.